pkg/network/netlink/conntracker.go

```diff
@@ -182,36 +217,25 @@ func (ctr *realConntracker) DeleteTranslation(c network.ConnectionStats) {
 		transport: c.Type,
 	}
 
-	t, ok := ctr.state[k]
-	if !ok {
-		log.Tracef("not deleting %+v: not found in conntrack cache", k)
-		return
+	if ctr.cache.Remove(k) {
+		atomic.AddInt64(&ctr.stats.unregisters, 1)
 	}
-
-	delete(ctr.state, k)
-	delete(ctr.state, ipTranslationToConnKey(k.transport, t))
-	atomic.AddInt64(&ctr.stats.unregisters, 1)
 }
 
 func (ctr *realConntracker) Close() {
 	ctr.consumer.Stop()
 	ctr.compactTicker.Stop()
-	ctr.exceededSizeLogLimit.Close()
 }
 
 func (ctr *realConntracker) loadInitialState(events <-chan Event) {
 	for e := range events {
 		conns := DecodeAndReleaseEvent(e)
 		for _, c := range conns {
-			if len(ctr.state) < ctr.maxStateSize && IsNAT(c) {
-				log.Tracef("%s", c)
-				if k, ok := formatKey(c.Origin); ok {
-					ctr.state[k] = formatIPTranslation(c.Reply)
-				}
-				if k, ok := formatKey(c.Reply); ok {
-					ctr.state[k] = formatIPTranslation(c.Origin)
-				}
+			if !IsNAT(c) {
+				continue
 			}
+
+			ctr.add(c, false)
```

Comment:
Can you explain why these initial entries don't need/want orphan tracking?

Reply:
This is so that long running non-active connections don't get marked as orphans. If these connections exist before we dump the conntrack table, and we mark them as orphans, there is a chance that they will never get accessed and so will get expunged from the cache after 2 minutes.

For entries that come in later, the assumption is that soon after the update from conntrack, there will be some activity, so they should get marked as non-orphans with the 2 minute TTL.